chia/full_node/block_store.py

```diff
@@ -553,16 +522,18 @@ async def get_block_records_by_hash(self, header_hashes: List[bytes32]) -> List[
             return []
 
         all_blocks: Dict[bytes32, BlockRecordDB] = {}
+        all_block_bytes: Dict[bytes32, memoryview] = {}
         if self.db_wrapper.db_version == 2:
             async with self.db_wrapper.reader_no_transaction() as conn:
                 async with conn.execute(
-                    "SELECT header_hash,block_record FROM full_blocks "
+                    "SELECT header_hash,block_record,block FROM full_blocks "
                     f'WHERE header_hash in ({"?," * (len(header_hashes) - 1)}?)',
                     header_hashes,
                 ) as cursor:
                     for row in await cursor.fetchall():
                         header_hash = bytes32(row[0])
                         all_blocks[header_hash] = BlockRecordDB.from_bytes(row[1])
+                        all_block_bytes[header_hash] = zstd.decompress(row[2])
```

Comment:
I think it would be better to just save off the plot filter info here, and maybe call the dictionary `all_plot_filter_info`
```
all_plot_filter_info[header_hash] = plot_filter_info_from_block(zstd.decompress(row[2]))
```
Also for the v1 case on line 552